src/metrics/matcher/namespaces.go

```diff
@@ -302,6 +315,13 @@ func (n *namespaces) process(value interface{}) error {
 					n.log.Error("failed to watch ruleset updates",
 						zap.String("ruleSetKey", ruleSet.Key()),
 						zap.Error(err))
+
+					// Track errors if we explicitly want to ensure watches succeed.
+					if n.requireNamespaceWatchOnInit {
```

Comment:
one awkward bit here is that this also covers the update case, not just the init case. However even if we propagate an error in that case, in the update path it gets swallowed regardless

Reply:
yeah makes sense